src/main/java/org/ohdsi/circe/cohortdefinition/builders/MeasurementSqlBuilder.java

```diff
@@ -138,6 +174,15 @@ protected List<String> resolveJoinClauses(T criteria) {
     if (criteria.providerSpecialty != null && criteria.providerSpecialty.length > 0) {
       joinClauses.add("LEFT JOIN @cdm_database_schema.PROVIDER PR on C.provider_id = PR.provider_id");
     }
+    if (criteria.measurementOperand != null && criteria.measurementOperand.measurement != null) {
+      BuilderOptions options = new BuilderOptions();
+      options.additionalColumns = Collections.singletonList(CriteriaColumn.VALUE_AS_NUMBER);
+      String subquery = getCriteriaSql((T) criteria.measurementOperand.measurement, options);
+      joinClauses.add(MessageFormat.format("JOIN (SELECT E.person_id, E.event_id, E.visit_occurrence_id, E.value_as_number, row_number() over (PARTITION BY E.person_id ORDER BY E.start_date, E.event_id {1}) as ordinal FROM (\n{0}\n) E) SC on C.person_id = SC.person_id",
+        subquery,
+        Optional.ofNullable(criteria.measurementOperand.limit).map(limit -> limit.equals("Last") ? "DESC" : "").orElse("")
+      ));
+    }
```

Comment:
Should there be time-bounding to looking for these measurements (either prior to or post the 'index event'). I believe you would want to specify this somehow, and maybe the extent of the prior/post time to select records from.    I understand 'same visit' addresses some of this, but I don't think you can depend on Measurement domains to come from the same visit, especially if the idea is to compare 2 measurements that could be spread across different visits.